modules/ztp-nbde-rekeying-all-nbde-nodes.adoc

```diff
@@ -0,0 +1,139 @@
+          exec:
+            command:
+            - cat
+            - /host/tmp/rekey-complete
+          initialDelaySeconds: 30
+          periodSeconds: 10
+        env:
+        - name: ROOT_DEV
+          value: /dev/disk/by-partlabel/root
+        - name: NEW_TANG_PIN
+          value: >-
+            {"t":1,"pins":{"tang":[
+              {"url":"http://tangserver01:7500","thp":"WOjQYkyK7DxY_T5pMncMO5w0f6E"},
+              {"url":"http://tangserver02:7500","thp":"I5Ynh2JefoAO3tNH9TgI4obIaXI"},
+              {"url":"http://tangserver03:7500","thp":"38qWZVeDKzCPG9pHLqKzs6k1ons"}
+            ]}}
+        volumeMounts:
+        - name: hostroot
+          mountPath: /host
+        securityContext:
+          privileged: true
+      volumes:
+      - name: hostroot
+        hostPath:
+          path: /
+      nodeSelector:
+        kubernetes.io/os: linux
+      priorityClassName: system-node-critical
+      restartPolicy: Always
+      serviceAccount: machine-config-daemon
+      serviceAccountName: machine-config-daemon
+----
++
+In this case, even though you are rekeying `tangserver01`, you must specify not only the new thumbprint for `tangserver01`, but also the current thumbprints for all other Tang servers.  Failure to specify all thumbprints for a rekeying operation opens up the opportunity for a man-in-the-middle attack.
+
+. Distribute the daemonset to every cluster that must be rekeyed. This can be as simple as running the following command:
++
+[source,terminal]
+----
+$ oc apply -f tang-rekey.yaml
+----
++
+However, to run at scale, wrap the daemonset in an ACM policy. This ACM configuration must contain one policy to deploy the daemonset,
+a second policy to check that all the daemonset pods are READY, and a placement rule to apply it to the appropriate set of clusters.
+
+[NOTE]
+====
+After validating that the daemonset has successfully rekeyed all servers, delete the daemonset. If you do not delete the daemonset, it must be deleted before the next rekeying operation.
+====
+
+.Verification
+
+After you distribute the daemonset, monitor the daemonsets to ensure that the rekeying has completed successfully. The script in the example daemonset terminates with an error if the rekeying failed, and remains in the `CURRENT` state if successful. There is also a readiness probe that marks the pod as `READY` when the rekeying has completed successfully.
+
+* This is an example of the output listing for the daemonset before the rekeying has completed:
++
+[source,terminal]
+----
+$ oc get -n openshift-machine-config-operator ds rekey-tang
+NAME         DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE
+rekey-tang   1         1         0       1            0           kubernetes.io/os=linux   11s
+----
+
+* This is an example of the output listing for the daemonset after the rekeying has completed successfully:
++
+[source,terminal]
+----
+$ oc get -n openshift-machine-config-operator ds rekey-tang
```

Comment:
The `oc` line is command, the rest is output.

Reply:
changed